### ISSUE TITLE
MidiMessage.noteOn() returns message with channel 1

### ISSUE THREAD

**Issue by vmoyankov** [patrickkidd/pyrtmidi#13]
Hi,

rtmidi.MidiMessage.noteOn() always return message for channel 1. rtmidi.MidiMessage.noteOff() is OK:

```
In [91]: rtmidi.MidiMessage.noteOn(10, 80, 127)
Out[91]: <NOTE ON, note: 80 (G#4), velocity: 127, channel: 1>

In [92]: m=rtmidi.MidiMessage.noteOn(10, 80, 127)

In [93]: m
Out[93]: <NOTE ON, note: 80 (G#4), velocity: 127, channel: 1>

In [94]: m.setChannel(10)

In [95]: m
Out[95]: <NOTE ON, note: 80 (G#4), velocity: 127, channel: 10>

In [96]: n=rtmidi.MidiMessage.noteOff(10,80)

In [97]: n
Out[97]: <NOTE OFF, note: 80 (G#4), channel: 10>
```

**Comment by patrickkidd** [comment]
Hmmm, that is strange, it’s working for me:

turin:dev patrick$ cat bleh.py
import rtmidi

m = rtmidi.MidiMessage.noteOn(10, 80, 126)
print(m)
turin:dev patrick$ python bleh.py
<NOTE ON, note: 80 (G#4), velocity: 126, channel: 10>
turin:dev patrick$ 

What version are you using, what OS, etc? What is your code?

**Comment by vmoyankov** [comment]
Hi,
Linux Mint, 18.2, python 3.5.2. pyrtmidi 2.3.2, installed with pip in virtualenv:

```
pip freeze
...
rtmidi==2.3.2
...
(pmidi)  ~/pmidi $ ipython
Python 3.5.2 (default, Nov 23 2017, 16:37:01) 
Type 'copyright', 'credits' or 'license' for more information
IPython 6.2.1 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import rtmidi

In [2]: rtmidi.MidiMessage.noteOn(10, 80, 127)
Out[2]: <NOTE ON, note: 80 (G#4), velocity: 127, channel: 1>

```

Here is the output from you snippet (in virtualenv, python3 again):

```
~/pmidi $ cat bleh.py 
import rtmidi

m = rtmidi.MidiMessage.noteOn(10, 80, 126)
print(m)

(pmidi)  ~/pmidi $ python3 bleh.py 
<NOTE ON, note: 80 (G#4), velocity: 126, channel: 1>
(pmidi)  ~/pmidi $ 
```

The same is with python 2.7.12 in virtualenv, :

```
$ virtualenv pimidi2
...
$ cd pimidi2/
~/pimidi2 $ . bin/activate
(pimidi2)  ~/pimidi2 $ pip install rtmidi
Collecting rtmidi
  Using cached rtmidi-2.3.2.tar.gz
Building wheels for collected packages: rtmidi
  Running setup.py bdist_wheel for rtmidi ... done
  Stored in directory: /home/XXXXX/.cache/pip/wheels/7c/70/76/415c8c79eaf938008c5fc74316b973c5f8ca30b88997e7a396
Successfully built rtmidi
Installing collected packages: rtmidi
Successfully installed rtmidi-2.3.2

(pimidi2) ~/pimidi2 $ ipython
WARNING: Attempting to work in a virtualenv. If you encounter problems, please install IPython inside the virtualenv.
Python 2.7.12 (default, Nov 20 2017, 18:23:56) 
Type "copyright", "credits" or "license" for more information.

IPython 2.4.1 -- An enhanced Interactive Python.
?         -> Introduction and overview of IPython's features.
%quickref -> Quick reference.
help      -> Python's own help system.
object?   -> Details about 'object', use 'object??' for extra details.

In [1]: import rtmidi

In [2]: m = rtmidi.MidiMessage.noteOn(10, 80, 126)

In [3]: print(m)
<NOTE ON, note: 80 (G#4), velocity: 126, channel: 1>

```

I didn't test it outside virtualenv. Let me know if I can help with troubleshooting this.
Regards,
Venko

**Comment by patrickkidd** [comment]
If there is problem anywhere it is in the bitwise functions in cpp_src/MidiMessage.cpp, which would make sense if this problem is bounded to a particular platform (linux/virtualenv).

To my knowledge other linux users have had success with this code for some time. I have not used linux in a while, as I am on mac and windows.

The code entry point begins at MidiMessage.cpp:522. The stack is not very deep for this call, feel free to debug and let me know.

-Patrick

**Comment by patrickkidd** [comment]
fixed in 007c0a56bea8a2e3577fbd09f06caf59c5faaa0e
